### ISSUE TITLE
d3.time.format.multi mutates array passed in

### ISSUE THREAD

**Issue by milroc** [mbostock/d3#1769]
I read through the API and it doesn't seem that this is the desired behavior. Is this the desired behavior?

```javascript
var formats = [
        [":%M:%S.%Lms", function(d) { return d.getMilliseconds(); }], // milliseconds
        [":%M:%Ss", function(d) { return d.getSeconds(); }], // seconds
        ["%-I:%M %p", function(d) { return d.getMinutes(); }], // minute
        ["%-I %p", function(d) { return d.getHours(); }], // hour
        ["%b %-d", function(d) { return d.getDay() && d.getDate() != 1; }], // day
        ["%b %-d", function(d) { return d.getDate() != 1; }], // monday of the week
        ["%Y %b", function(d) { return d.getMonth(); }], // month
        ["%Y", function() { return true; }] // year
      ];
formats.forEach(function(d) { console.log(typeof d[0]); });
// (8) string
d3.time.format.multi(formats);
formats.forEach(function(d) { console.log(typeof d[0]); });
// (8) function
```

**Comment by mbostock** [comment]
This is going away in 4.0, so moving this to the icebox. See the new multi-scale time format in the [d3-time documentation](https://github.com/d3/d3-time-format).

**Comment by mbostock** [comment]
Closing as fixed in d3-time-format for 4.0.
